app/controllers/concerns/checkout_callbacks.rb

```diff
@@ -47,7 +50,25 @@ def load_saved_credit_cards
   end
 
   def load_shipping_methods
-    @shipping_methods = available_shipping_methods.sort { |a, b| a.name.casecmp(b.name) }
+    @shipping_methods = sorted_available_shipping_methods
+  end
+
+  def load_allowed_shipping_methods
+    @allowed_shipping_methods = sorted_available_shipping_methods.filter(
+      &method(:supports_all_products_shipping_categories?)
+    )
+  end
```

Comment:
This is called a second time from the `#edit` method. We should cache the ivar.
```suggestion
  def load_allowed_shipping_methods
    @allowed_shipping_methods ||= sorted_available_shipping_methods.filter(
      &method(:supports_all_products_shipping_categories?)
    )
  end
```